### PR TITLE
add example for setting up grafana dashboard

### PR DESCRIPTION
**Pull request by anushkamittal2001**
## Related issue #
N/A (recently setup grafana with kind cluster and struggled a bit)

## Proposed Changes
Add an example of setting up kyverno with grafana in a kind clusyter

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/website/blob/main/CONTRIBUTING.md).
- [x] I have inspected the website preview for accuracy.
- [x] I have signed off my issue.

### CODE REVIEW

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 1]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
 ![Dashboard example 1](dashboard-example-1.png)
 
 ![Dashboard example 2](dashboard-example-2.png)
+
+### Example: Setup Grafana with Kyverno and a Kind cluster
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:44]:
```suggestion
### Tutorial
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 2]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+kind create cluster
+```
+
+Add helm repositories
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:50]:
```suggestion
Add Helm repositories.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 3]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts 
+helm repo add kyverno https://kyverno.github.io/kyverno/ 
+```
+
+Update repos
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:57]:
```suggestion
Update Helm repositories.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 4]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+helm repo update    
+```
+
+Install Kyverno and monitoring
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:63]:
```suggestion
Install Kyverno and the kube-prometheus-stack Helm chart.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 5]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+helm install kyverno kyverno/kyverno --namespace kyverno --create-namespace 
+helm install monitoring prometheus-community/kube-prometheus-stack -n monitoring --create-namespace
+
+k -n monitoring get po -l "release"
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:69]:
```suggestion
kubectl -n monitoring get po -l "release"
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 6]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+```
+
+Create the service monitor
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:72]:
```suggestion
Create the ServiceMonitor resource.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 7]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+`vi sm.yaml`
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:74]:
```suggestion
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 8]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+Add the following service monitor yaml
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:76]:
```suggestion
Add the following contents to a file named `service-monitor.yaml`.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 9]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+k label ns kyverno app.kubernetes.io/instance=kyverno                                                               
+k label ns kyverno app.kubernetes.io/name=kyverno
+```
+
+Apply service-monitor.yaml
+
+```sh
+k apply -f sm.yaml
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:115]:
```suggestion
kubectl apply -f service-monitor.yaml
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 10]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+```
+
+Add right labels
+
+```sh
+k label ns kyverno app.kubernetes.io/instance=kyverno                                                               
+k label ns kyverno app.kubernetes.io/name=kyverno
+```
+
+Apply service-monitor.yaml
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:112]:
```suggestion
Create the ServiceMonitor.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 11]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+      insecureSkipVerify: true
+  namespaceSelector:
+    matchNames:
+    - kyverno
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: kyverno
+```
+
+Add right labels
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:105]:
```suggestion
Add the appropriate labels.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 12]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+k label ns kyverno app.kubernetes.io/instance=kyverno                                                               
+k label ns kyverno app.kubernetes.io/name=kyverno
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:109]:
```suggestion
kubectl label ns kyverno app.kubernetes.io/instance=kyverno                                                               
kubectl label ns kyverno app.kubernetes.io/name=kyverno
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 13]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+```
+
+Apply service-monitor.yaml
+
+```sh
+k apply -f sm.yaml
+```
+
+Restart deployments and sts in monitoring ns
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:118]:
```suggestion
Restart Deployments and StatefulSets in the `monitoring` Namespace.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 14]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+k rollout restart deploy,sts -n monitoring
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:121]:
```suggestion
kubectl rollout restart deploy,sts -n monitoring
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 15]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+```
+
+Check services in monitoring ns
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:124]:
```suggestion
Check services in the `monitoring` Namespace.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 16]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+k get svc -n monitoring
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:127]:
```suggestion
kubectl get svc -n monitoring             
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 17]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+Check services in monitoring ns
+
+```sh
+k get svc -n monitoring             
+                                                                                                            
+NAME                                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
+monitoring-kube-prometheus-prometheus     ClusterIP   10.96.238.189   <none>        9090/TCP,8080/TCP            4h16m
+```
+
+Port forward the svc/monitoring-kube-prometheus-prometheus to a local port
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:133]:
```suggestion
Port forward the `monitoring-kube-prometheus-prometheus` Service to a local port.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 18]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+sudo kubectl port-forward svc/monitoring-kube-prometheus-prometheus 81:9090 -n monitoring
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:136]:
```suggestion
kubectl port-forward svc/monitoring-kube-prometheus-prometheus 81:9090 -n monitoring                                                       
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 19]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+Port forward the svc/monitoring-kube-prometheus-prometheus to a local port
+
+```sh
+sudo kubectl port-forward svc/monitoring-kube-prometheus-prometheus 81:9090 -n monitoring                                                       
+
+Forwarding from 127.0.0.1:81 -> 9090
+Forwarding from [::1]:81 -> 9090
+```
+
+Similarly port forward the svc/monitoring-grafana to another local port
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:142]:
```suggestion
Similarly, port forward the `monitoring-grafana` Service to another local port.
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 20]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+```sh
+k get svc -n monitoring
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:145]:
```suggestion
kubectl get svc -n monitoring               
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 21]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+Similarly port forward the svc/monitoring-grafana to another local port
+
+```sh
+k get svc -n monitoring               
+                                                                             
+NAME                                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
+monitoring-grafana                        ClusterIP   10.96.188.20    <none>        80/TCP                       4h19m
+
+sudo kubectl port-forward svc/monitoring-grafana -n monitoring 82:80
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:150]:
```suggestion
kubectl port-forward svc/monitoring-grafana -n monitoring 82:3000
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 22]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+NAME                                      TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)                      AGE
+monitoring-grafana                        ClusterIP   10.96.188.20    <none>        80/TCP                       4h19m
+
+sudo kubectl port-forward svc/monitoring-grafana -n monitoring 82:80    
+
+Forwarding from 127.0.0.1:82 -> 3000
+Forwarding from [::1]:82 -> 3000
+```
+
+Go to Prometheus on port 81 (in this case) and check status -> Targets -> Filter for kyverno (to see if metrics are getting scraped)
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:156]:
```suggestion
Go to Prometheus on port 9090 and check status -> Targets -> Filter for kyverno (to see if metrics are getting scraped)
```

File: content/en/docs/monitoring/bonus-grafana-dashboard/index.md [hunk 23]
```diff
@@ -40,3 +40,123 @@ curl -fsS https://raw.githubusercontent.com/kyverno/kyverno/main/charts/kyverno/
...
+
+Go to Grafana on port 82 (in this case) -> Dashboards -> New ->  import -> Upload file that you get from running the below command -> Data type = Prometheus -> import
```
Comment (chipzoller) [inline content/en/docs/monitoring/bonus-grafana-dashboard/index.md:158]:
```suggestion
Go to Grafana on port 3000 -> Dashboards -> New ->  import -> Upload file that you get from running the below command -> Data type = Prometheus -> import
```

**Review by chipzoller — CHANGES_REQUESTED** [review]
Hi @anushkamittal2001, see these proposed changes. Some general guidelines which will help in writing good documentation:

1. Don't use abbreviations or aliased commands. `k` may not be aliased for all users and so you should always prefer the formal, full program name `kubectl`.
2. Ensure your commands match up to the results. When you have inconsistencies in commands versus output, it's confusing for users who may not have your level of expertise.
3. Be consistent in how you write.

**Comment by chipzoller** [comment]
@anushkamittal2001, awaiting changes here

**Comment by anushkamittal2001** [comment]
@chipzoller 

Thank you for all your suggestions, I have noted the guidelines you shared for the future! I have merged all your suggestions. Please let me know if it looks okay.

**Review by chipzoller — APPROVED** [review]

**Pull request merged**